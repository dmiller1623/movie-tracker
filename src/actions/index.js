export const populateMovies = (movies) => ({
  type: 'POPULATE_MOVIES',
  movies
});

export const loginUser = (user) => ({
  type: 'LOGIN_USER',
  user
});

export const signUpUser = (email, password) => ({
  type: 'SIGNUP_USER',
  email,
  password
});

export const signOutUser = () => {
  type: 'SIGN_OUT'

}
