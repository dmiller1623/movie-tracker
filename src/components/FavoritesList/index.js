import React from 'react';
import MovieCard from '../../containers/MovieCard';

const FavoritesList = ({ movies, favorites }) => {
  const displayedMovies = favorites.map( (favorite, index) => {
    const movie = movies.find( movie => movie.movie_id === favorite);
    return <MovieCard {...movie} key={index} starColor={'#ffd24d'} />;
  });
  
  return (
    <main>
      {displayedMovies}
    </main>
  );
};

export default FavoritesList;
