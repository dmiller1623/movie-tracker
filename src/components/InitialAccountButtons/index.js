import React from 'react';

export const InitialAccountButtons = ({handleLogin, handleSignUp}) => {

  return (
    <div>
        <button
          onClick={() => handleLogin()}>
          Login
        </button>
        <button
          onClick={() => handleSignUp()}>
          Sign Up
        </button>
    </div>
  );
};
