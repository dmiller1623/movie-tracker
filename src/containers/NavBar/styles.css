nav {
  margin: 2rem 0 .5rem 0; }

@media only screen and (max-width: 100px) {
  nav {
    display: flex;
    flex-direction: column; } }
