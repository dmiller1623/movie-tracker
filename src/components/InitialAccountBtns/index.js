import React from 'react';
import { NavLink } from 'react-router-dom';

import './styles.css';

export const InitialAccountBtns = () => {
  return (
    <div>
      <NavLink activeClassName='selected' className='initialBtns' to='/login'>
        Login
      </NavLink>
      <span> | </span>
      <NavLink activeClassName='selected' className='initialBtns' to='/signup'>
        Sign Up
      </NavLink>
    </div>
  );
};
