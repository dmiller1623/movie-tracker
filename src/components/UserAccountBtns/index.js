import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import './styles.css';

const UserAccountBtns = ({ name, handleSignOut }) => {
  return (
    <div className='userBtns'>
      <h3>Hi, {name}!</h3>
      <NavLink activeClassName='selected' className='favoritesBtn' to='/favorites'>
        View Favorites
      </NavLink>
      <span> | </span>
      <button onClick={() => handleSignOut()}>
        Sign Out
      </button>
    </div>
  )
}

export default UserAccountBtns;

UserAccountBtns.propTypes = {
  name: PropTypes.string,
  handleSignOut: PropTypes.func
};
