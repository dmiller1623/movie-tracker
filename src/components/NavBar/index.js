import React, { Component } from 'react';
import Login from '../../containers/Login';
import { InitialAccountButtons } from '../../components/InitialAccountButtons';
import { connect } from 'react-redux'
import UserAccountBtns from '../UserAccountBtns';
import { signOutUser } from '../../actions';
import SignUp from '../../containers/SignUp';


class NavBar extends Component {
  constructor(){
    super();
    this.state = {
      pageStatus: ''
    };
  }
  
  handleLogin = () => {
    this.setState({ pageStatus: 'LOGIN'})
  }

  handleSignUp = () => {
    this.setState({ pageStatus: 'SIGNUP'})
  }

  handleSubmit = (signOut) => {
    if (signOut) {
      this.setState({ pageStatus: 'INITIAL'})
    } else { 
      this.setState({ pageStatus: 'LOGGED_IN'})
    }
  } 

  render() {
    let navDisplay;
    switch(this.state.pageStatus) {
      case ('LOGIN'):
      navDisplay = <Login handleSubmit={this.handleSubmit} />;
      break;
      case ('SIGNUP'):
      navDisplay = <SignUp handleSubmit={this.handleSubmit} />;
      break;
      case ('LOGGED_IN'):
      navDisplay =  <UserAccountBtns name={this.props.user.name} signOutUser={this.props.signOutUser} handleSubmit={this.handleSubmit} />;
      break;
      default:
      navDisplay = <InitialAccountButtons handleLogin={this.handleLogin}  handleSignUp={this.handleSignUp}/>;
      break;
    }
    let userBtns;
    
    return (
      <div>
        {navDisplay}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user
});

const mapDispatchToProps = dispatch => ({
  signOutUser: () => dispatch(signOutUser())
});

export default connect(mapStateToProps, mapDispatchToProps, null, {pure: false})(NavBar)