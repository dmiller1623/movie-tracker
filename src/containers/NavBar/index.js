import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { InitialAccountBtns } from '../../components/InitialAccountBtns';
import UserAccountBtns from '../../components/UserAccountBtns';
import { signOutUser } from '../../actions';
import PropTypes from 'prop-types';

import './styles.css';

export class NavBar extends Component {
  handleSignOut = () => {
    this.props.signOutUser();
    localStorage.clear();
    this.props.history.push('/');
  }

  render() {
    const { name } = this.props.user;
    const navDisplay = name ? 
      <UserAccountBtns name={name} handleSignOut={this.handleSignOut}/> :
      <InitialAccountBtns />;
    
    return (
      <nav>
        {navDisplay}
      </nav>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
});

export const mapDispatchToProps = (dispatch) => ({
  signOutUser: () => dispatch(signOutUser())
});

NavBar.propTypes = {
  signOutUser: PropTypes.func,
  user: PropTypes.object,
  history: PropTypes.object
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(NavBar));
