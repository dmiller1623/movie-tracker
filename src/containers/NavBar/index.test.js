import React from 'react'
import { shallow, mount } from 'enzyme'
import {  NavBar, mapStateToProps, mapDispatchToProps } from '../NavBar'
import { signOutUser } from '../../actions'
import { InitialAccountButtons } from '../../components/InitialAccountButtons';
import UserAccountBtns from '../../components/UserAccountBtns';

describe('NavBar', () => {
  describe('NavBar component', () => {
    let wrapper
    it('shouls match the snapshot', () => {
      const user= {
        name: 'Dennis',
        email: 'dmill@aol.com',
        password: 'password'
      }
      wrapper = shallow(<NavBar user={user}/>)
      expect(wrapper).toMatchSnapshot()
    })

    it('should call the function signOutUser when handle sign out is ran', () => {
      const history = []
      const mockSignOut = jest.fn()
      const user= {
        name: 'Dennis',
        email: 'dmill@aol.com',
        password: 'password'
      }
      wrapper = shallow(<NavBar user={user} signOutUser={mockSignOut} history={history}/>)
      wrapper.instance().handleSignOut()
      expect(mockSignOut).toHaveBeenCalled()
    })

    it('should render user account buttons if a user is signed in', () =>{
      const user = {
        name: 'dennis'
      }
      wrapper = shallow(<NavBar user={user}/>)
      expect(wrapper.find(UserAccountBtns).length).toEqual(1)
    })

    it('should render initial account buttons if no user is signed in', () => {
      const user = {}

      wrapper = shallow(<NavBar user={user}/>)
      expect(wrapper.find(InitialAccountButtons).length).toEqual(1)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function MDTP', () => {

      const mockDispatch = jest.fn()
      const actionToDispatch = signOutUser()

      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.signOutUser()

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)

    })
  })

  describe('mapStateToProps', () => {
    it('should return a user object', () => {
      const mockState = {
        user: {},
        favorites: []
      }
      const expected = {
        user: {}
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})