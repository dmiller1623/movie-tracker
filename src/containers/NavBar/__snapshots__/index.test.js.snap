// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar NavBar component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    user={
        Object {
            "email": "dmill@aol.com",
            "name": "Dennis",
            "password": "password",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <UserAccountBtns
        handleSignOut={[Function]}
        name="Dennis"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "handleSignOut": [Function],
        "name": "Dennis",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <UserAccountBtns
          handleSignOut={[Function]}
          name="Dennis"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleSignOut": [Function],
          "name": "Dennis",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
